packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -251,6 +251,52 @@ describe('CDK Include', () => {
     );
   });
 
+  test("correctly parses templates with parameters", () => {
+    const cfnTemplate = includeTestTemplate(stack, 'bucket-with-parameters.json');
+    const param = cfnTemplate.getParameter('BucketName');
+    const bucket = new s3.CfnBucket(stack, "newBucket", {"bucketName": param.valueAsString});
+
+    expect(stack).toMatchTemplate(
+      {
+        "Resources": {
+          "Bucket": {
+            "Type": "AWS::S3::Bucket",
+            "Properties": {
+              "BucketName": {
+                "Ref": "BucketName",
+              },
+            },
+          },
+          "newBucket": {
+            "Type": "AWS::S3::Bucket",
+            "Properties": {
+              "BucketName": {
+                "Ref": "BucketName",
+              },
+            },
+          },
+        },
+        "Parameters": {
+          "BucketName": {
+            "Type": "String",
+            "Default": "MyS3Bucket",
+            "Description": "The name of your bucket",
+          },
+        },
+      },
+    );
+
+    expect(bucket.bucketName).toBeDefined();
```

Comment:
This will always be define (`bucket.bucketName` has type `string`), so this assertion doesn't add anything. I would just remove it.